### ISSUE TITLE
cfn-lint shows E3510 with conditional entry in IAM policy

### ISSUE THREAD

**Issue by ilikejam** [aws-cloudformation/cfn-lint#3727]
### CloudFormation Lint Version

cfn-lint 1.15.1

### What operating system are you using?

Mac and Linux

### Describe the bug

Hi.

cfn-lint fails with E3510 if an entry in an IAM policy is wrapped in a Fn::If

E3510 [{'Action': ['s3:GetObject*'], 'Effect': 'Allow', 'Resource': [{'Fn::Join': ['/', [{'Fn::FindInMap': ['Environment', {'Ref': 'Environment'}, 'MyProdBucket']}, '*']]}]}] is not of type 'object'
bug.yml:50:17

The AWS console appears to be happy with the template.

### Expected behavior

cfn-lint should not show E3510

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Description: cfn-lint bug

Parameters:
  Environment:
    Type: String
    Default: test
    AllowedValues:
      - test
      - prod
    Description: Environment this stack represents

Mappings:
  Environment:
    prod:
      MyBucket: "arn:aws:s3:::my-prod-bucket"
      MyProdBucket: "arn:aws:s3:::my-only-in-prod-bucket"
    test:
      MyBucket: "arn:aws:s3:::my-test-bucket"
      MyProdBucket: dummy

Conditions:
  IsProdEnvironment: !Equals
    - Ref: Environment
    - prod

Resources:
  MyUser:
    Type: "AWS::IAM::User"
    Properties:
      UserName: !Sub ${Environment}-myuser

  MyUserPolicy:
    Type: "AWS::IAM::Policy"
    Properties:
      PolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Action:
              - "s3:GetObject*"
            Resource:
              - "Fn::FindInMap":
                  - Environment
                  - Ref: Environment
                  - MyBucket
          - "Fn::If":
              - IsProdEnvironment
              - - Action:
                    - "s3:GetObject*"
                  Effect: Allow
                  Resource:
                    - "Fn::Join":
                        - /
                        - - "Fn::FindInMap":
                              - Environment
                              - Ref: Environment
                              - MyProdBucket
                          - "*"
              - Ref: "AWS::NoValue"
      PolicyName: MyUserPolicy
      Users:
        - "Ref:MyUser"
```

**Comment by ilikejam** [comment]
Closing - ignore. cfn-lint was right!
